### DISCUSSION TITLE
multiple authors are not displayed

### DISCUSSION THREAD

**Discussion by jphconstantin** [mmistakes/minimal-mistakes#2564] [Q&A]
## Environment

- Minimal Mistakes version:
- Ruby gem or remote theme version:
- Jekyll version: 4.1.0
- Git repository URL:
- GitHub Pages hosted (if yes provide URL to site):
- Operating system:

## Other

It seems that only the first author among a list is displayed ... am I wrong ?
In the font matter (page or post), i set:
author_profile: true
authors:
- author_1
- author_2
I created an authors.yml file in _data

**Comment by iBug** [comment]
It's the intended behavior. The ability of loading multiple authors from `data.yml` only makes it more convenient to show different authors for different posts, not multiple authors on the same post.

**Comment by jphconstantin** [comment]
Are you sure ?
I don't see the interest to indicate a list of authors in a post IF only the first one is displayed !!!

**Comment by iBug** [comment]
If you mean the left sidebar, then yes for reasons of simplicity. Specifying multiple authors may be useful elsewhere.

**Comment by jphconstantin** [comment]
Yes I mean the sidebar.
Wait for the position of the author. I forgot perhaps a parameter.

**Comment by iBug** [comment]
You're not missing anything

<https://github.com/mmistakes/minimal-mistakes/blob/776ab26ce11e786bbce5c632e53036e9e670344c/_includes/author-profile.html#L1>

If you have `authors` specified then only the first one is displayed on the left.

**Comment by jphconstantin** [comment]
Could we consider that as a bug ?

**Comment by iBug** [comment]
Not a bug I would say. It's intentional (or the code wouldn't have been written like this). Displaying multiple authors in the current appearance is kinda ugly IMO.

**Comment by mmistakes** [comment]
As @iBug has stated, this is intended behavior. Only one author will be shown on a post, page, document, etc. The theme supports "multiple" authors in the sense that you can have `Author A` assigned to Post 1, and `Author B` assigned to Post 2.

Not `Author A` and `Author B` on Post 1.

**Comment by jphconstantin** [comment]
Could you explain me how to display the list of authors in the sidebar independently of all considerations ? that could be an acceptable solution for me ...

**Comment by mmistakes** [comment]
There is no mechanism to do this without you hacking up the theme.
You'd need to loop through `page.authors` or something similar and spit out author blocks that way.

Have a look through the closed issues. This has come up before and pretty sure I offered suggestions on how to achieve it.

**Comment by jphconstantin** [comment]
I tried this but that doesn't work

{% for author in site.data.authors %}
    {{ author.name }} -- {{ author.email }}
{% endfor %}

**Comment by iBug** [comment]
You'll need to look into the code and tweak accordingly, instead of blindly trying random snippets retrieved elsewhere without understanding them. As a starting point `_includes/author-profile.html` is the file you should be investigating.

**Comment by mmistakes** [comment]
https://github.com/mmistakes/minimal-mistakes/issues/1341
